pkg/network/ebpf/c/protocols/events.h

```diff
@@ -0,0 +1,118 @@
+#ifndef __USM_EVENTS_H
+#define __USM_EVENTS_H
+
+#include "events-types.h"
+#include "bpf_telemetry.h"
+
+/* USM_EVENTS_INIT defines two functions used for the purposes of buffering and sending
+   data to userspace:
+   1) <name>_batch_enqueue
+   1) <name>_flush_batch
+   For more information of this please refer to
+   pkg/networks/protocols/events/README.md */
+#define USM_EVENTS_INIT(name, value, batch_size)                                        \
+    _Static_assert((sizeof(value)*batch_size) <= BATCH_BUFFER_SIZE,                     \
+                   _STR(name)" batch is too large");                                    \
+                                                                                        \
+    BPF_PERCPU_ARRAY_MAP(name##_batch_state, __u32, batch_state_t, 1)                   \
+    BPF_PERF_EVENT_ARRAY_MAP(name##_batch_events, __u32, 0)                             \
+    BPF_HASH_MAP(name##_batches, batch_key_t, batch_data_t, 0)                          \
+                                                                                        \
+    static __always_inline bool name##_batch_full(batch_data_t *batch) {                \
+        return batch && batch->len == batch_size;                                       \
+    }                                                                                   \
```

Comment:
I think that function can be shared among all protocols

Reply:
only if `batch_size` is turned into an argument. I don't see much of a benefit :shrug: in either case so I'd just leave as it is